Partial support for Braze catalogs

hugolepetit opened the pull request:
This adds partial support for [Braze catalogs](https://www.braze.com/docs/api/endpoints/catalogs)

I don't have the approved company bandwith ™️ to build everything for those endpoints, I focused on : 
- Catalog management endpoints
- Catalog items endpoints

I ended up adding `delete` and `put` verbs to `BrazeRuby::HTTP`. You might want to have extra scrutiny here
I did my best to stick to the code style already present in the rest of the Gem but might be off, i will happily address any concern you have to get this merged

Thanks a lot for your time 🙏

jonallured commented:
Hi @hugolepetit thanks for your PR! Can you please rebase on latest main? I've been working on upgrading/updating the gem today and so a lot changed since you opened this one. 😅

hugolepetit commented:
@jonallured Sure thing, i will try to update the README.MD to reflect the new supported endpoints as well

jonallured commented:
Cool, thanks! It would also be great to have the CHANGELOG updated.

This PR is failing on CircleCI because it has Standard Ruby violations - I bet you could get it to green with something like this:

```
$ bundle exec standardrb --fix
```

hugolepetit commented:
@jonallured 
The standardrb command worked like a charm. The CI is now green, Thanks!
Added some README documentation, I saw other pieces of documentation mentioning the 50 items max per request, did not know if we should repeat or have an indication at top of documentation to gather this. While at it we could point out to general api rate limits braze documentation. Since you will probably proofread things, I will let you choose what you feel is best.
I also renamed create_catalog -> create_catalogs since it supports a list of catalogs
We should bit one step closer to a merge now. Appreciate your feedback on all those changes 🙇

hugolepetit commented:
CHANGELOG updated as well

jonallured commented on CHANGELOG.md:
```diff
@@ -11,6 +11,13 @@ The format is based on [Keep a Changelog][] and this project adheres to
+    * delete_catalog
+    * create_catalog_items
+    * update_catalog_items
+    * delete_catalog_items
```
This can be simplified like so:

```
* Partial support for Braze catalogs ([#60][])
```

And then down towards the bottom please add:

```
[#61]: https://github.com/jonallured/braze_ruby/pull/61
```

jonallured reviewed (commented):
Almost there!! The only code change is a simplification of the CHANGELOG so that it matches the other entries. For this I just copy/paste the title of the PR and then link to it.

Then my other ask is that you rebase this on latest main so that the git history is linear and there is no merge commit. If you'd like to keep the individual commits that's fine but you could also squash them down into one. Up to you!

hugolepetit commented:
I've done a rebase and took the opportunity to merge together a bunch of related commits to avoid commits such as `cleanup` or `fix standardrb warnings`.

Sure the add partial support for Braze catalogs is a bit meaty but the structure is pretty much the same for the different endpoints so a reader should not be too lost. 
I also picked some of the context previously contained in the atomic commits to add some depth to its description

Regarding the add of the PR reference towards the end of the CHANGELOG, i did not append it to the others (chronological) but went with a sorted list of PRs because of a preference bias towards readability.

I hope this wraps it up 🙏

jonallured commented:
EPIC 🎉

hugolepetit commented:
Ahhh that warms my heart @jonallured. Thanks a lot!
